common/configuration/asm_follower.go

```diff
@@ -51,7 +53,14 @@ func (f *asmFollower) sync(ctx context.Context, values *xsync.MapOf[Ref, SyncedV
 		IncludeValues: &includeValues,
 	}))
 	if err != nil {
-		return fmt.Errorf("error getting secrets list from leader: %w", err)
+		if f.errorFilter.ReportLeaseError() {
```

Comment:
I do think this is too general, like was mentioned on this morning's call. What @alecthomas said about responses that include errors vs network errors feels like a good way to differentiate.

Reply:
Do we have an example somewhere of how to determine the type of error? The library docs mention connect.Error but that does not seem to match what is in the datadog logs.

I never thought I would miss Java checked exceptions, but they do make situations like this easier.

Reply:
Maybe something like this: https://github.com/TBD54566975/ftl/blob/main/backend/controller/ingress/handler.go#L55-L62

Reply:
Thanks, I have updated it to use the return codes to decide on if it should attempt to filter